src/clarity/debug/dap/mod.rs

```diff
@@ -0,0 +1,1146 @@
+                                body: Some(ResponseBody::Evaluate(EvaluateResponse {
+                                    result: map_type.clone(),
+                                    result_type: Some(map_type),
+                                    presentation_hint: Some(VariablePresentationHint {
+                                        kind: Some(VariableKind::Data),
+                                        attributes: Some(vec![
+                                            VariableAttribute::HasDataBreakpoint,
+                                        ]),
+                                        visibility: None,
+                                        lazy: None,
+                                    }),
+                                    variables_reference: 0,
+                                    named_variables: None,
+                                    indexed_variables: None,
+                                    memory_reference: None,
+                                })),
+                            }
+                        } else {
+                            Response {
+                                request_seq: seq,
+                                success: false,
+                                message: Some("undefined".to_string()),
+                                body: None,
+                            }
+                        }
+                    }
+                    Err(e) => Response {
+                        request_seq: seq,
+                        success: false,
+                        message: Some(e),
+                        body: None,
+                    },
+                }
+            }
+            _ => match self
+                .get_state()
+                .evaluate(env, context, &arguments.expression)
+            {
+                Ok(value) => Response {
+                    request_seq: seq,
+                    success: true,
+                    message: None,
+                    body: Some(ResponseBody::Evaluate(EvaluateResponse {
+                        result: value.to_string(),
+                        result_type: Some(type_for_value(&value)),
+                        presentation_hint: None,
+                        variables_reference: 0,
+                        named_variables: None,
+                        indexed_variables: None,
+                        memory_reference: None,
+                    })),
+                },
+                Err(errors) => Response {
+                    request_seq: seq,
+                    success: false,
+                    message: Some(errors.join("\n")),
+                    body: None,
+                },
+            },
+        };
+
+        self.send_response(response);
+        false
+    }
+
+    fn quit(&mut self, seq: i64, arguments: DisconnectArguments) -> bool {
+        // match arguments.restart {
+        //     Some(restart) => restart,
+        //     None => false,
+        // }
```

Comment:
Dead code?

Reply:
yes. I'll add an issue to support restart later too.